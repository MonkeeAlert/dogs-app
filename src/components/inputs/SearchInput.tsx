import {StyleSheet, Text, TextInput, View} from 'react-native';
import React from 'react';
import {Icon} from 'react-native-elements/dist/icons/Icon';
import {colors} from '../../utils/constants';

interface Props {
  value: string;
  action?: (str: string) => void;
  placeholder?: string;
  isDisabled?: boolean;
  isAutofocused?: boolean;
}

const ICON_SIZE = 36;

const SearchInput = ({
  value = '',
  action,
  placeholder,
  isDisabled,
  isAutofocused,
}: Props) => {
  const isPlaceholderForDisabledInputPresent = value === '' && placeholder;

  return (
    <View style={styles.container}>
      <Icon
        size={21}
        name={'search'}
        type={'ionicon'}
        style={styles.icon}
        color={colors.gray}
      />
      {isDisabled ? (
        <View testID={'SearchInput_Disabled'} style={styles.input}>
          <Text
            style={
              isPlaceholderForDisabledInputPresent
                ? styles.placeholderText
                : {color: colors.black}
            }>
            {isPlaceholderForDisabledInputPresent ? placeholder : value}
          </Text>
        </View>
      ) : (
        <TextInput
          style={styles.input}
          value={value}
          placeholder={placeholder}
          placeholderTextColor={colors.gray}
          testID={'SearchInput_Input'}
          onChangeText={action}
          autoFocus={isAutofocused}
        />
      )}
    </View>
  );
};

export default SearchInput;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignSelf: 'flex-end',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: 999,
    overflow: 'hidden',
    backgroundColor: colors.white,
    marginVertical: 7,
  },
  icon: {
    width: ICON_SIZE,
    height: ICON_SIZE,
    borderRadius: ICON_SIZE,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
    left: 7,
  },
  input: {
    flex: 1,
    padding: 7,
    marginRight: 14,
    color: colors.black,
    height: 36,
    justifyContent: 'center',
  },
  placeholderText: {
    color: colors.gray,
  },
});
