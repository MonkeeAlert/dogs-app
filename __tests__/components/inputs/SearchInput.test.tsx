import {shallow} from 'enzyme';
import React from 'react';
import {create} from 'react-test-renderer';
import SearchInput from '../../../src/components/inputs/SearchInput';

describe('SearchInput', () => {
  it('matches snapshot', () => {
    const tree = create(<SearchInput />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled view', () => {
    const wrapper = shallow(<SearchInput isDisabled={true} />);
    const view = wrapper.find({testID: 'SearchInput_Disabled'});

    expect(view.exists()).toBe(true);
  });

  it('renders disabled view with placeholder same as view', () => {
    const wrapper = shallow(
      <SearchInput isDisabled={true} placeholder={'foo'} value={''} />,
    );
    const view = wrapper.find({testID: 'SearchInput_Disabled'});

    expect(view.exists()).toBe(true);
  });
});
