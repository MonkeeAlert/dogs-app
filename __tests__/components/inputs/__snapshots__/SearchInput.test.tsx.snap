// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-end",
      "backgroundColor": "#fffafb",
      "borderRadius": 999,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 7,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 36,
          "height": 36,
          "justifyContent": "center",
          "left": 7,
          "position": "relative",
          "width": 36,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#bbb",
                "fontSize": 21,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          ï”…
        </Text>
      </View>
    </View>
  </View>
  <TextInput
    placeholderTextColor="#bbb"
    style={
      Object {
        "color": "#131515",
        "flex": 1,
        "height": 36,
        "justifyContent": "center",
        "marginRight": 14,
        "padding": 7,
      }
    }
    value=""
  />
</View>
`;
